server/src/parser/regex.ts

```diff
@@ -0,0 +1,10 @@
+export const NUMERIC_REGEX = /^[+-]?\d+\.?\d*(([eE][-+])?\d+)*/;
```

Comment:
Can we go over these to make sure they are accurate and cover edge cases?

Reply:
Yes, should add unittests for known edge-cases.